test/benchgc/src/benchgc/tuner/op_config.py

```diff
@@ -0,0 +1,175 @@
+
+import json
+import math
+import os
+
+from gc_mlir.extras import types as T
+from gc_mlir.ir import IntegerAttr, OpView
+
+
+class Config:
+    def __init__(self):
+        self.field_candidates = {}
+        self.field_constraints = {}
+        self.init_candidates()
+        self.init_constraints()
+
+    def init_candidates(self):
+        pass
+
+    def init_constraints(self):
+        pass
+
+    def attach_to_ir(self, op: OpView):
+        pass
+
+
+def find_factors(num):
+    factors = set()
+    for i in range(1, int(math.sqrt(num)) + 1):
+        if num % i == 0:
+            factors.add(i)
+            factors.add(num // i)
+    return sorted(factors)
+
+
+class MatMulConfig(Config):
+    def __init__(
+        self,
+        op: OpView = None,
+        M_threads: int = 1,
+        K_threads: int = 1,
+        N_threads: int = 1,
+        M_block: int = 1,
+        K_block: int = 1,
+        N_block: int = 1,
+        innermostM_block: int = 1,
+        innermostK_block: int = 1,
+        innermostN_block: int = 1,
+    ):
+        # you can set the default value and candidates by info from matmul_op
+        self.M = op.inputs[0].type.shape[0]
+        self.K = op.inputs[0].type.shape[1]
+        self.N = op.inputs[1].type.shape[1]
+        # self.input_a_dtype = str(op.inputs[0].type.element_type)
+        self.num_threads = int(os.environ.get("OMP_NUM_THREADS", 1))
+        self.M_threads = M_threads
+        self.K_threads = K_threads
+        self.N_threads = N_threads
+        self.M_block = M_block
+        self.K_block = K_block
+        self.N_block = N_block
+        self.innermostM_block = innermostM_block
+        self.innermostK_block = innermostK_block
+        self.innermostN_block = innermostN_block
+        super().__init__()
+
+    def init_candidates(self):
+        default_blocks = [16, 32, 64, 128, 256, 512]
+        default_innermost_blocks = [16, 32]
+        self.field_candidates["M_threads"] = find_factors(self.num_threads)
+        self.field_candidates["K_threads"] = find_factors(self.num_threads)
+        self.field_candidates["N_threads"] = find_factors(self.num_threads)
+        self.field_candidates["M_block"] = [
+            block for block in default_blocks if self.M >= block
+        ]
+        self.field_candidates["K_block"] = [
+            block for block in default_blocks if self.K >= block
+        ]
+        self.field_candidates["N_block"] = [
+            block for block in default_blocks if self.N >= block
+        ]
+        self.field_candidates["innermostM_block"] = [
+            block for block in default_innermost_blocks if self.M >= block
+        ]
+        self.field_candidates["innermostK_block"] = [
+            block for block in default_innermost_blocks if self.K >= block
+        ]
+        self.field_candidates["innermostN_block"] = [
+            block for block in default_innermost_blocks if self.N >= block
+        ]
```

Comment:
It is better to provide the grid options by command line. Developer can control the search space in this way.